client/consensus/grandpa/lib.go

```diff
@@ -24,3 +26,37 @@ type NewAuthoritySetStruct[H comparable, N constraints.Unsigned] struct {
 	SetId       N
 	Authorities []Authority
 }
+
+type msgData[H comparable, N constraints.Unsigned] struct {
```

Comment:
I would say to avoid short names for struct since they can be confused with variable names
```suggestion
type messageData[H comparable, N constraints.Unsigned] struct {
```